waddrmgr/address.go

```diff
@@ -113,6 +116,10 @@ var _ ManagedPubKeyAddress = (*managedAddress)(nil)
 // used by the caller without worrying about it being zeroed during an address
 // lock.
 func (a *managedAddress) unlock(key EncryptorDecryptor) ([]byte, error) {
+	if a.WatchingOnly() {
+		return nil, managerError(ErrWatchingOnly, errWatchingOnly, nil)
```

Comment:
Don't think so, I see `unlock` being called from `scriptAddress.Script` and `managedAddress.PrivKey` both of which start with the check `if a.WatchingOnly() return nil, managerError(ErrWatchingOnly, errWatchingOnly, nil)`